src/main/java/com/google/sps/servlets/TaskServlet.java

```diff
@@ -39,84 +28,90 @@
-        }
-
-        // Check that the input is within the requested range.
-        if (rewardPts < 0 || rewardPts > 200) {
-            System.err.println("User input is out of range: " + rewardPtsString);
-            return -1;
-        }
-
-        return rewardPts;
+    // If the input is nonempty and valid, set the taskDetail value to the input value
+    if (!taskDetail.equals("")) {
+      long creationTime = System.currentTimeMillis();
+
+      // Create an Entity that stores the input comment
+      Entity taskEntity = new Entity("Task");
+      taskEntity.setProperty("detail", taskDetail);
+      taskEntity.setProperty("timestamp", creationTime);
+      taskEntity.setProperty("reward", rewardPts);
+      taskEntity.setProperty("status", "OPEN");
+      taskEntity.setProperty("Owner", "Leonard");
+      taskEntity.setProperty("Helper", "N/A");
+      taskEntity.setProperty("Address", "4xxx Cxxxxx Avenue, Pittsburgh, PA 15xxx");
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(taskEntity);
+    } else {
+      System.err.println("The input task detail is empty");
+      response.sendError(
+          HttpServletResponse.SC_FORBIDDEN, "The task detail field cannot be empty.");
+      return;
     }
 
-    @Override
-    public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String keyString = request.getParameter("key");
+    // Redirect back to the user page.
+    response.sendRedirect("/user_profile.html");
+  }
+
+  /** Return the input rewarding points by the user, or -1 if the input was invalid */
+  private int getRewardingPoints(HttpServletRequest request, String inputName) {
+    // Get the input from the form.
+    String rewardPtsString = request.getParameter(inputName);
+
+    // Convert the input to an int.
+    int rewardPts;
+    try {
+      rewardPts = Integer.parseInt(rewardPtsString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + rewardPtsString);
+      return -1;
+    }
 
-        Key taskKey = KeyFactory.stringToKey(keyString);
+    // Check that the input is within the requested range.
+    if (rewardPts < 0 || rewardPts > 200) {
+      System.err.println("User input is out of range: " + rewardPtsString);
+      return -1;
+    }
 
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        datastore.delete(taskKey);
+    return rewardPts;
+  }
 
-        // Redirect to the user profile page
-        response.sendRedirect("/user_profile.html");
-    }
-}
\ No newline at end of file
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    String keyString = request.getParameter("key");
+
+    Key taskKey = KeyFactory.stringToKey(keyString);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.delete(taskKey);
```

Comment:
Will you leave a TODO comment here for error handling? We should handle the case where the user attempts to delete a non-existent task.

Reply:
Oh sure, I will add a TODO comment here and handle the exceptional case in a new PR.